pkg/userdata/centos/provider.go

```diff
@@ -268,11 +268,16 @@ write_files:
   content: |
 {{ containerRuntimeHealthCheckSystemdUnit | indent 4 }}
 
-{{- if .InsecureRegistries }}
+{{- if or .InsecureRegistries .RegistryMirrors }}
 - path: /run/containers/registries.conf
   permissions: "0644"
   content: |
-    INSECURE_REGISTRY="--insecure-registry {{ .InsecureRegistries | join " --insecure-registry "}}"
+    {{- if .InsecureRegistries}}
+    INSECURE_REGISTRY="{{range .InsecureRegistries}}--insecure-registry {{.}} {{end}}"
```

Comment:
Why did you change this expression? The prior one did not leave a trailing space.

Reply:
Henrik asked me to change it to a loop.

Reply:
Yes, that `strings.Join` was rather counter intuitive & meant we always set `--insecure-registry`.
That worked well as long as we only had one variable - Since we now have 2 which can be set individually that would not have worked